NewKernel_d/test/NewKernel_d/Epick_d.cpp

```diff
@@ -1,6 +1,7 @@
 #include <boost/config.hpp>
 
 //#define BOOST_RESULT_OF_USE_DECLTYPE 1
+#define CGAL_DO_NOT_USE_BOOST_MP // this test does not compile with cpp_rational
```

Comment:
@mglisse Could you please have a look at this [compilation error](https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-5.4-Ic-83/NewKernel_d/TestReport_lrineau_Ubuntu-GCC_master_CXX20-Release.gz) if you have time? It is happenning only when `cpp_rational` type from `boost mp` is selected. For other types including `mpq_rational` it works. I tried to fix it but I cannot find a good solution because I do not fully understand what exactly causes this error and what is missing inside `cpp_rational` that is not missing inside `mpq_rational`. There is only one error of this type and it is inside `NewKernel_d`. To continue testing this PR and avoiding errors in the github workflow, I switch off `Epick_d.cpp` test for the `cpp_rational` type, but it is an ugly solution and I would like to avoid it. Thank you in advance!

Reply:
I'll take a look, but I can't make any promise on how soon...
Random try, since this involves Eigen, does `#include <boost/multiprecision/eigen.hpp>` have any effect?

Reply:
Just tried. Nope. I will also continue searching for better solutions.

Reply:
https://github.com/boostorg/multiprecision/issues/393
(it is a bit strange to see `-DCGAL_USE_GMPXX=1` in the compiler arguments for something that ends up using boost multiprecision)
This error only seems to appear with very recent versions of Eigen.

Reply:
Thanks for having a look into that! Regarding compiler arguments, this is a temporary hack that I will remove. It is there in order to test this boost type and `to_interval` implementation in the regular CGAL test suite. Once tested, I will clean it up. This is the unchecked task in the description of this PR.

Reply:
@mglisse:
> (it is a bit strange to see `-DCGAL_USE_GMPXX=1` in the compiler arguments for something that ends up using boost multiprecision)

The definition of `CGAL_USE_GMPXX` is triggered by our CMake scripts. So far, this PR has not touched anything in CMake, as far as I know: you patch tweaks `Number_types/include/CGAL/Exact_integer.h`, reversing the logic so that `CGAL_USE_BOOST_MP` has a precedence over `CGAL_USE_GMPXX`.

Let us remember that `CGAL_USE_BOOST_MP` is defined that way:
```c++
#if !defined CGAL_DO_NOT_USE_BOOST_MP && \
    (!defined _MSC_VER || BOOST_VERSION >= 107000)
#define CGAL_USE_BOOST_MP 1
```
https://github.com/danston/cgal/blob/d8da3502bc8feb9a4bfbcf3bc369bfd77f53e8b0/Number_types/include/CGAL/boost_mp.h#L17-L24

and then in this PR you have add in `Number_types/include/CGAL/Exact_integer.h`:
```c++
#if BOOST_VERSION > 107700 && defined(CGAL_USE_BOOST_MP)
// TODO: That is used for testing, it must be removed when merging into master.
{ typedef BOOST_cpp_arithmetic_kernel::Rational Type; };
#else // BOOST_VERSION <= 107700
#ifdef CGAL_USE_GMPXX
{ typedef mpq_class Type; };
#elif defined(CGAL_USE_GMP)
```
https://github.com/danston/cgal/blob/b2f6c0c5df43032b8c8ea0363048ce257e14d0c0/Number_types/include/CGAL/Number_types/internal/Exact_type_selector.h#L58-L64

So with the current patch, `CGAL_USE_GMPXX` is only considered if `CGAL_DO_NOT_USE_BOOST_MP` was defined by the user, or `BOOST_VERSION <= 107700`.

That shows the big issue with the naming of preprocessor macros `CGAL_USE_<something>`: they are used by CMake to instruct the C++ code that `<something>` was found by CMake and *can be used*. The naming is not right. And we are not consistent.
  - for TBB, that macro is named `CGAL_LINKED_WITH_TBB`,
  - for Eigen, it is named `CGAL_EIGEN3_ENABLED`,
  - and for GMPXX, it is `CGAL_USE_GMPXX`...
  
We should choose something and stick with it.

Reply:
This line generated a warning [here](https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-5.4-Ic-109/NewKernel_d/TestReport_cgaltester_x86-64_Darwin-18.5_Apple-clang-default_Release.gz)

Reply:
@sloriot I think this conversion can be closed.